src/main/java/io/antmedia/muxer/HLSMuxer.java

```diff
@@ -209,12 +206,38 @@ public String getOutputURL()
 		return super.getOutputURL();
 	}
 
+	public String getCustomHeaderStr() {
+		String customHeaderStr = "";
+
+		if (mainTrackId != null) {
+			customHeaderStr += "mainTrackId: " + mainTrackId + "\r\n";
+		}
+
+		if (streamId != null) {
+			customHeaderStr += "streamId: " + streamId + "\r\n";
+		}
+
+		return customHeaderStr;
+	}
 
 	public AVFormatContext getOutputFormatContext() {
 		if (outputFormatContext == null) {
 
 			outputFormatContext= new AVFormatContext(null);
 			int ret = avformat_alloc_output_context2(outputFormatContext, null, format, getOutputURL());
+
+			if((StringUtils.isNotBlank(httpEndpoint))) {
+				AVDictionary options = new AVDictionary();
+				String customHeaderStr = getCustomHeaderStr();
+				if(StringUtils.isNotBlank(customHeaderStr)){
+
+					av_dict_set(options, "headers", customHeaderStr, 0);
```

Comment:
Please don't use http headers because it makes the implementation more complex. 

Just append the generated subfolder path to the end of `httpEndpoint`. In this way, likely you don't need to change anything in `UploadHLSChunk`

Reply:
I have some concerns regarding to that
`httpEndpoint + getExtendedS3StreamsFolderPath(mainTrackId, streamId, getAppSettings().getS3StreamsFolderPath())`

I think we can pass mainTrackId, streamId to UploadHLSChunk this way but i think this might break backward compatability because by default
getAppSettings().getS3StreamsFolderPath()  is "streams" and on UploadHLSChunk its added at the end of the path

so users are settings this through
s3StreamsFolderPath app setting
suppose it is set like:
`"streams/%m/%s"`

lets say we sent it like this:
`http://localhost:5080/WebRTCAppEE/hls-upload/testroom/yunus
`
would end up like `streams/%m/%s/testroom/yunus` because UploadHlsChunk has
`appSettings.getS3StreamsFolderPath() + File.separator + req.getPathInfo()
`

Reply:
I understand the problem. 

I think all these problems are coming because you use `s3StreamsFolderPath`. The point in the solution is to creaet a generic solution for subfolder structure. So please create a subFolder settings in the AppSettings and use it.

Fortunately, Broadcast object has also subFolder settings and that parameters passed to the muxers. You'll just need to change `StringUtils.isNotBlank(broadcast.getSubFolder) ? broadcast.getSubFolder : appSettings.getSubFolder`  in the muxer's init methods

Reply:
If I add an extra subFolder setting in the app settings, it will result in two settings with similar names, which might confuse users. While this approach simplifies the code, it comes at the cost of making the app less user-friendly/confusing. Had a discusson with @burak-58 about  this and he also thought this way.

Reply:
TBH, If you think two settings make the app less user-friendly/confusing, I can say that they are totally different things. 

`s3StreamsFolderPath` is about S3 root folder where the streams are recorded. `subfolder` settings are applied to streams not only for S3, it's for local disk recording as well because users are asking for these kinds of settings from time to time. 

In our first discussion for this implementation, please correct me if I'm wrong. I remember that we agreed on having another setting for this implementation and adding this setting to the end of `httpEndpoint` to provide a generic solution. Then you've implemented in another way. I'm wondering what the reason is that you proceed in a different way.